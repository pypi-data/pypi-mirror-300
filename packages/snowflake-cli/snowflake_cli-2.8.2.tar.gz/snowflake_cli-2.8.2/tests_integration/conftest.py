# Copyright (c) 2024 Snowflake Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from __future__ import annotations

import functools
import json
import shutil
import tempfile
from contextlib import contextmanager
from dataclasses import dataclass
from json import JSONDecodeError
from pathlib import Path
from typing import Any, Dict, List, Optional

import pytest
import yaml

from snowflake.cli.api.cli_global_context import cli_context_manager
from snowflake.cli.app.cli_app import app_factory
from typer import Typer
from typer.testing import CliRunner

from tests.conftest import clean_logging_handlers_fixture  # noqa: F401
from tests.testing_utils.fixtures import (
    alter_snowflake_yml,  # noqa: F401
    snowflake_home,
)
from tests.testing_utils.files_and_dirs import merge_left

pytest_plugins = [
    "tests_common",
    "tests_integration.testing_utils",
    "tests_integration.snowflake_connector",
]

TEST_DIR = Path(__file__).parent
DEFAULT_TEST_CONFIG = "connection_configs.toml"


@dataclass
class CommandResult:
    exit_code: int
    json: Optional[List[Dict[str, Any]] | Dict[str, Any]] = None
    output: Optional[str] = None
    stderr: Optional[str] = None


class TestConfigProvider:
    def __init__(self, temp_dir_with_configs: Path):
        self._temp_dir_with_configs = temp_dir_with_configs

    def get_config_path(self, file_name: str) -> Path:
        return self._temp_dir_with_configs / file_name


@pytest.fixture(scope="session")
def test_snowcli_config_provider():
    with tempfile.TemporaryDirectory() as td:
        temp_dst = Path(td) / "config"
        shutil.copytree(TEST_DIR / "config", temp_dst)
        yield TestConfigProvider(temp_dst)


@pytest.fixture(scope="session")
def test_root_path():
    return TEST_DIR


class SnowCLIRunner(CliRunner):
    def __init__(self, app: Typer, test_config_provider: TestConfigProvider):
        super().__init__()
        self.app = app
        self._test_config_provider = test_config_provider
        self._test_config_path = self._test_config_provider.get_config_path(
            DEFAULT_TEST_CONFIG
        )

    def use_config(self, config_file_name: str) -> None:
        self._test_config_path = self._test_config_provider.get_config_path(
            config_file_name
        )

    @functools.wraps(CliRunner.invoke)
    def invoke(self, *a, **kw):
        if "catch_exceptions" not in kw:
            kw.update(catch_exceptions=False)
        return super().invoke(self.app, *a, **kw)

    def invoke_with_config(self, args, **kwargs) -> CommandResult:
        result = self.invoke(
            [
                "--config-file",
                self._test_config_path,
                *args,
            ],
            **kwargs,
        )

        if result.output == "" or result.output.strip() == "Done":
            return CommandResult(result.exit_code, json=[])
        try:
            return CommandResult(
                result.exit_code, json.loads(result.output), output=result.output
            )
        except JSONDecodeError:
            return CommandResult(result.exit_code, output=result.output)

    def invoke_json(self, args, **kwargs) -> CommandResult:
        return self.invoke_with_config([*args, "--format", "JSON"], **kwargs)

    def invoke_with_connection_json(
        self, args, connection: str = "integration", **kwargs
    ) -> CommandResult:
        return self.invoke_json([*args, "-c", connection], **kwargs)

    def invoke_with_connection(
        self, args, connection: str = "integration", **kwargs
    ) -> CommandResult:
        return self.invoke_with_config([*args, "-c", connection], **kwargs)


@pytest.fixture
def runner(test_snowcli_config_provider):
    app = app_factory()
    yield SnowCLIRunner(app, test_snowcli_config_provider)


class QueryResultJsonEncoderError(RuntimeError):
    def __init__(self, output: str):
        super().__init__(f"Can not parse query result:\n{output}")


@pytest.fixture
def project_directory(temporary_working_directory, test_root_path):
    @contextmanager
    def _temporary_project_directory(
        project_name,
        merge_project_definition: Optional[dict] = None,
        subpath: Optional[Path] = None,
    ):
        test_data_file = test_root_path / "test_data" / "projects" / project_name
        project_dir = temporary_working_directory
        if subpath:
            project_dir = temporary_working_directory / subpath
            project_dir.mkdir(parents=True)
        shutil.copytree(test_data_file, project_dir, dirs_exist_ok=True)
        if merge_project_definition:
            with Path("snowflake.yml").open("r") as fh:
                project_definition = yaml.safe_load(fh)
            merge_left(project_definition, merge_project_definition)
            with open(Path(project_dir) / "snowflake.yml", "w") as file:
                yaml.dump(project_definition, file)

        yield project_dir

    return _temporary_project_directory


@pytest.fixture(autouse=True)
def reset_global_context_after_each_test(request):
    cli_context_manager.reset()
    yield


# This automatically used fixture isolates default location
# of config files from user's system.
@pytest.fixture(autouse=True)
def isolate_snowflake_home(snowflake_home):
    yield snowflake_home
