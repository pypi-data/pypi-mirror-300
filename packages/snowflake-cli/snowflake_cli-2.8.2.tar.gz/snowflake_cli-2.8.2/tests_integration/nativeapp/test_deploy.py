# Copyright (c) 2024 Snowflake Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import os
import uuid

import yaml

from snowflake.cli.api.project.util import generate_user_env


from tests.nativeapp.utils import touch

from tests.project.fixtures import *
from tests_integration.test_utils import (
    contains_row_with,
    not_contains_row_with,
    pushd,
    row_from_snowflake_session,
    enable_definition_v2_feature_flag,
)
from tests_integration.testing_utils import (
    assert_that_result_failed_with_message_containing,
)

USER_NAME = f"user_{uuid.uuid4().hex}"
TEST_ENV = generate_user_env(USER_NAME)


def sanitize_deploy_output(output):
    deploy_root = Path("output/deploy").resolve()
    return output.replace(USER_NAME, "@@USER@@").replace(
        str(deploy_root), "@@DEPLOY_ROOT@@"
    )


# Tests a simple flow of executing "snow app deploy", verifying that an application package was created, and an application was not
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy(
    definition_version,
    project_directory,
    runner,
    snowflake_session,
    snapshot,
    print_paths_as_posix,
):
    project_name = "myapp"
    with project_directory(f"napp_init_{definition_version}"):
        result = runner.invoke_with_connection(
            ["app", "deploy"],
            env=TEST_ENV,
        )
        assert result.exit_code == 0
        assert sanitize_deploy_output(result.output) == snapshot

        try:
            # package exist
            package_name = f"{project_name}_pkg_{USER_NAME}".upper()
            app_name = f"{project_name}_{USER_NAME}".upper()
            assert contains_row_with(
                row_from_snowflake_session(
                    snowflake_session.execute_string(
                        f"show application packages like '{package_name}'",
                    )
                ),
                dict(name=package_name),
            )

            # manifest file exists
            stage_name = "app_src.stage"  # as defined in native-apps-templates/basic
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(stage_files.json, {"name": "stage/manifest.yml"})

            # app does not exist
            assert not_contains_row_with(
                row_from_snowflake_session(
                    snowflake_session.execute_string(
                        f"show applications like '{app_name}'",
                    )
                ),
                dict(name=app_name),
            )

            # re-deploying should be a no-op; make sure we don't issue any PUT commands
            result = runner.invoke_with_connection_json(
                ["app", "deploy", "--debug"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0
            assert "Successfully uploaded chunk 0 of file" not in result.output

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize(
    "command,contains,not_contains",
    [
        # deploy --prune removes remote-only files
        [
            "app deploy --prune --no-validate",
            ["stage/manifest.yml"],
            ["stage/README.md"],
        ],
        # deploy removes remote-only files (--prune is the default value)
        ["app deploy --no-validate", ["stage/manifest.yml"], ["stage/README.md"]],
        # deploy --no-prune does not delete remote-only files
        ["app deploy --no-prune", ["stage/README.md"], []],
    ],
)
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_prune(
    command,
    contains,
    not_contains,
    definition_version,
    project_directory,
    runner,
    snapshot,
    print_paths_as_posix,
):
    project_name = "myapp"
    with project_directory(f"napp_init_{definition_version}"):
        result = runner.invoke_with_connection_json(
            ["app", "deploy"],
            env=TEST_ENV,
        )
        assert result.exit_code == 0

        try:
            # delete a file locally
            os.remove(os.path.join("app", "README.md"))

            # deploy
            result = runner.invoke_with_connection(
                command.split(),
                env=TEST_ENV,
            )
            assert result.exit_code == 0
            assert sanitize_deploy_output(result.output) == snapshot

            # verify the file does not exist on the stage
            package_name = f"{project_name}_pkg_{USER_NAME}".upper()
            stage_name = "app_src.stage"  # as defined in native-apps-templates/basic
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            for name in contains:
                assert contains_row_with(stage_files.json, {"name": name})
            for name in not_contains:
                assert not_contains_row_with(stage_files.json, {"name": name})

            # make sure we always delete the app
            result = runner.invoke_with_connection_json(
                ["app", "teardown"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests a simple flow of executing "snow app deploy [files]", verifying that only the specified files are synced to the stage
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_files(
    definition_version,
    project_directory,
    runner,
    snapshot,
    print_paths_as_posix,
):
    project_name = "myapp"
    with project_directory(f"napp_init_{definition_version}"):
        # sync only two specific files to stage
        result = runner.invoke_with_connection(
            [
                "app",
                "deploy",
                "app/manifest.yml",
                "app/setup_script.sql",
                "--no-validate",
            ],
            env=TEST_ENV,
        )
        assert result.exit_code == 0
        assert sanitize_deploy_output(result.output) == snapshot

        try:
            # manifest and script files exist, readme doesn't exist
            package_name = f"{project_name}_pkg_{USER_NAME}".upper()
            stage_name = "app_src.stage"  # as defined in native-apps-templates/basic
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(stage_files.json, {"name": "stage/manifest.yml"})
            assert contains_row_with(
                stage_files.json, {"name": "stage/setup_script.sql"}
            )
            assert not_contains_row_with(stage_files.json, {"name": "stage/README.md"})

            # make sure we always delete the app
            result = runner.invoke_with_connection_json(
                ["app", "teardown"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that files inside of a symlinked directory are deployed
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_nested_directories(
    definition_version,
    project_directory,
    runner,
    snapshot,
    print_paths_as_posix,
):
    project_name = "myapp"
    with project_directory(f"napp_init_{definition_version}"):
        # create nested file under app/
        touch("app/nested/dir/file.txt")

        result = runner.invoke_with_connection(
            ["app", "deploy", "app/nested/dir/file.txt", "--no-validate"],
            env=TEST_ENV,
        )
        assert result.exit_code == 0
        assert sanitize_deploy_output(result.output) == snapshot

        try:
            package_name = f"{project_name}_pkg_{USER_NAME}".upper()
            stage_name = "app_src.stage"  # as defined in native-apps-templates/basic
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(
                stage_files.json, {"name": "stage/nested/dir/file.txt"}
            )

            # make sure we always delete the app
            result = runner.invoke_with_connection_json(
                ["app", "teardown"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that deploying a directory recursively syncs all of its contents
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_directory(
    definition_version,
    project_directory,
    runner,
):
    project_name = "myapp"
    with project_directory(f"napp_init_{definition_version}"):
        touch("app/dir/file.txt")
        result = runner.invoke_with_connection(
            ["app", "deploy", "app/dir", "--no-recursive", "--no-validate"],
            env=TEST_ENV,
        )
        assert_that_result_failed_with_message_containing(
            result, "Add the -r flag to deploy directories."
        )

        result = runner.invoke_with_connection(
            ["app", "deploy", "app/dir", "-r", "--no-validate"],
            env=TEST_ENV,
        )
        assert result.exit_code == 0

        try:
            package_name = f"{project_name}_pkg_{USER_NAME}".upper()
            stage_name = "app_src.stage"  # as defined in native-apps-templates/basic
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(stage_files.json, {"name": "stage/dir/file.txt"})

            # make sure we always delete the app
            result = runner.invoke_with_connection_json(
                ["app", "teardown"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that deploying a directory without specifying -r returns an error
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_directory_no_recursive(
    definition_version,
    project_directory,
    runner,
):
    with project_directory(f"napp_init_{definition_version}"):
        try:
            touch("app/nested/dir/file.txt")
            result = runner.invoke_with_connection_json(
                ["app", "deploy", "app/nested", "--no-validate"],
                env=TEST_ENV,
            )
            assert result.exit_code == 1, result.output

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that specifying an unknown path to deploy results in an error
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_unknown_path(
    definition_version,
    project_directory,
    runner,
):
    with project_directory(f"napp_init_{definition_version}"):
        try:
            result = runner.invoke_with_connection_json(
                ["app", "deploy", "does_not_exist", "--no-validate"],
                env=TEST_ENV,
            )
            assert result.exit_code == 1
            assert "The following path does not exist:" in result.output

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that specifying a path with no deploy artifact results in an error
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_path_with_no_mapping(
    definition_version,
    project_directory,
    runner,
):
    with project_directory(f"napp_init_{definition_version}"):
        try:
            result = runner.invoke_with_connection_json(
                ["app", "deploy", "snowflake.yml", "--no-validate"],
                env=TEST_ENV,
            )
            assert result.exit_code == 1
            assert "No artifact found for" in result.output

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that specifying a path and pruning result in an error
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_rejects_pruning_when_path_is_specified(
    definition_version,
    project_directory,
    runner,
):
    with project_directory(f"napp_init_{definition_version}"):
        try:
            os.unlink("app/README.md")
            result = runner.invoke_with_connection_json(
                ["app", "deploy", "app/README.md", "--prune"],
                env=TEST_ENV,
            )
            assert result.exit_code == 1
            assert (
                "--prune cannot be used when paths are also specified" in result.output
            )

        finally:
            # teardown is idempotent, so we can execute it again with no ill effects
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that specifying a path with no direct mapping falls back to search for prefix matches
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_looks_for_prefix_matches(
    definition_version,
    project_directory,
    runner,
    snapshot,
    print_paths_as_posix,
):
    project_name = "myapp"

    with project_directory(f"napp_deploy_prefix_matches_{definition_version}"):
        try:
            result = runner.invoke_with_connection(
                ["app", "deploy", "-r", "app"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0
            assert sanitize_deploy_output(result.output) == snapshot

            package_name = f"{project_name}_pkg_{USER_NAME}".upper()
            stage_name = "app_src.stage"  # as defined in native-apps-templates/basic
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(stage_files.json, {"name": "stage/manifest.yml"})
            assert contains_row_with(
                stage_files.json, {"name": "stage/setup_script.sql"}
            )
            assert contains_row_with(stage_files.json, {"name": "stage/README.md"})
            assert not_contains_row_with(
                stage_files.json, {"name": "stage/src/main.py"}
            )
            assert not_contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/c/c.py"}
            )

            result = runner.invoke_with_connection(
                ["app", "deploy", "-r", "lib/parent/child/c"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/c/c.py"}
            )
            assert not_contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/a.py"}
            )
            assert not_contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/b.py"}
            )

            result = runner.invoke_with_connection(
                ["app", "deploy", "lib/parent/child/a.py"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/c/c.py"}
            )
            assert contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/a.py"}
            )
            assert not_contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/b.py"}
            )

            result = runner.invoke_with_connection(
                ["app", "deploy", "lib", "-r"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/c/c.py"}
            )
            assert contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/a.py"}
            )
            assert contains_row_with(
                stage_files.json, {"name": "stage/parent-lib/child/b.py"}
            )

        finally:
            result = runner.invoke_with_connection(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0


# Tests that snow app deploy -r . deploys all changes
@pytest.mark.integration
@enable_definition_v2_feature_flag
@pytest.mark.parametrize("definition_version", ["v1", "v2"])
def test_nativeapp_deploy_dot(
    definition_version,
    project_directory,
    runner,
    snapshot,
    print_paths_as_posix,
):
    project_name = "myapp"
    with project_directory(f"napp_init_{definition_version}"):
        try:
            result = runner.invoke_with_connection(
                ["app", "deploy", "-r", "."],
                env=TEST_ENV,
            )
            assert result.exit_code == 0
            assert sanitize_deploy_output(result.output) == snapshot

            package_name = f"{project_name}_pkg_{USER_NAME}".upper()
            stage_name = "app_src.stage"  # as defined in native-apps-templates/basic
            stage_files = runner.invoke_with_connection_json(
                ["stage", "list-files", f"{package_name}.{stage_name}"],
                env=TEST_ENV,
            )
            assert contains_row_with(stage_files.json, {"name": "stage/manifest.yml"})
            assert contains_row_with(
                stage_files.json, {"name": "stage/setup_script.sql"}
            )
            assert contains_row_with(stage_files.json, {"name": "stage/README.md"})

        finally:
            result = runner.invoke_with_connection_json(
                ["app", "teardown", "--force"],
                env=TEST_ENV,
            )
            assert result.exit_code == 0
