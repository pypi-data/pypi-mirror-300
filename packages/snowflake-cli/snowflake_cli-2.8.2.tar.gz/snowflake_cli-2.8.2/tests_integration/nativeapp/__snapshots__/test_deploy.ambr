# serializer version: 1
# name: test_nativeapp_deploy[v1]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/README.md -> README.md
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy[v2]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/README.md -> README.md
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_dot[v1]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/README.md -> README.md
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_dot[v2]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/README.md -> README.md
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_files[v1]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_files[v2]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_looks_for_prefix_matches[v1]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/README.md -> README.md
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_looks_for_prefix_matches[v2]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/README.md -> README.md
    added:    app/manifest.yml -> manifest.yml
    added:    app/setup_script.sql -> setup_script.sql
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_nested_directories[v1]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/nested/dir/file.txt -> nested/dir/file.txt
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_nested_directories[v2]
  '''
  Creating new application package myapp_pkg_@@USER@@ in account.
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Local changes to be deployed:
    added:    app/nested/dir/file.txt -> nested/dir/file.txt
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_prune[v1-app deploy --no-prune-contains2-not_contains2]
  '''
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  The following files exist only on the stage:
  README.md
  
  Use the --prune flag to delete them from the stage.
  Your stage is up-to-date with your local deploy root.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_prune[v1-app deploy --no-validate-contains1-not_contains1]
  '''
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Deleted paths to be removed from your stage:
    deleted:  README.md
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_prune[v1-app deploy --prune --no-validate-contains0-not_contains0]
  '''
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Deleted paths to be removed from your stage:
    deleted:  README.md
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_prune[v2-app deploy --no-prune-contains2-not_contains2]
  '''
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  The following files exist only on the stage:
  README.md
  
  Use the --prune flag to delete them from the stage.
  Your stage is up-to-date with your local deploy root.
  Validating Snowflake Native App setup script.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_prune[v2-app deploy --no-validate-contains1-not_contains1]
  '''
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Deleted paths to be removed from your stage:
    deleted:  README.md
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
# name: test_nativeapp_deploy_prune[v2-app deploy --prune --no-validate-contains0-not_contains0]
  '''
  Checking if stage myapp_pkg_@@USER@@.app_src.stage exists, or creating a new one if none exists.
  Performing a diff between the Snowflake stage and your local deploy_root ('@@DEPLOY_ROOT@@') directory.
  Deleted paths to be removed from your stage:
    deleted:  README.md
  Updating the Snowflake stage from your local @@DEPLOY_ROOT@@ directory.
  Deployed successfully. Application package and stage are up-to-date.
  
  '''
# ---
